### ISSUE TITLE
Batch API: Change RequestEntityTooLargeException from Internal to Public

### ISSUE THREAD

**Issue by gorillapower** [Azure/azure-cosmos-dotnet-v3#889]
We are trying to handle 413 requests (PayloadTooLarge) when executing Batch requests. We encountered scenarios where the batch was over the 2mb limit, but instead of checking the size of each batch on each request, we thought to handle the error optimistically by catching the `DocumentClientException` or similar when it occurs.

It seems that for Batch operations, the SDK throws `Microsoft.Azure.Documents.RequestEntityTooLargeException` when the 2mb limit is reached, however, the class internal and cannot be caught. 

![image](https://user-images.githubusercontent.com/8879065/66558512-f4008b00-eb53-11e9-8631-50c8c345e54c.png)

Can this exception be exposed publicly so we can have the option to handle such exceptions?

The workaround at the moment is to catch by `Exception` and inspect the message to see if its indeed a 'RequestEntityTooLargeException', which isnt ideal.

Open to alternative suggestions on how to handle the 2mb limit efficiently as well.

**Comment by j82w** [comment]
@gorillapower can you not catch a CosmosException with status code 413? The v3 SDK should not be throwing DocumentClientExceptions anywhere.

**Comment by gorillapower** [comment]
Hi @j82w, I saw the base type of the `Exception` was `DocumentClientException` when I was debugging, which is why i mentioned DocumentClientException - they type was 'Microsoft.Azure.Documents.RequestEntityTooLargeException'.

I just tested with trying to catch `CosmosException` and nothing was caught.

If i look at the class, it seems to be coming from `Assembly Microsoft.Azure.DocumentDB.Core, Version=2.7.0.0`, Perhaps this has something to do with the fact that I'm running the `DocumentDB` and the `CosmosDB` SDKs in parallel?

*running 
<PackageReference Include="Microsoft.Azure.Cosmos" Version="3.2.0-preview2" />
<PackageReference Include="Microsoft.Azure.DocumentDB.Core" Version="2.7.0" />
